pkg/kclient/services.go

```diff
@@ -0,0 +1,36 @@
+package kclient
+
+import (
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// CreateService generates and creates the service
+// commonObjectMeta is the ObjectMeta for the service
+func (c *Client) CreateService(commonObjectMeta metav1.ObjectMeta, svcSpec corev1.ServiceSpec) (*corev1.Service, error) {
+	svc := corev1.Service{
+		ObjectMeta: commonObjectMeta,
+		Spec:       svcSpec,
+	}
+
+	service, err := c.KubeClient.CoreV1().Services(c.Namespace).Create(&svc)
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to create Service for %s", commonObjectMeta.Name)
```

Comment:
is wrapping error is necessary here?? IMO we should return bare errors from client functions, as they are intended to be called my multiple functions and it helps in checking conditions based on error codes.

Reply:
I'm following the error format for all the other functions under pkg/kclient (pod.go, volumes.go, deployments.go).

Reply:
ya I see, resolving it for your PR :), will raise a separate concern.